docs/doxygen/collectives.md

```diff
@@ -1,20 +1,150 @@
+and other information required for the collective operation. All participants of
+the team should call the initialize operation. The routine returns once the
+participants enter the collective initialize operation. The collective operation
+is invoked using a ucc\_collective\_post operation.
+ucc\_collective\_init\_and\_post operation initializes as well as post the
+collective operation.
 
-\b Collective type: The collective operation supported by UCC is defined by the enumeration ucc\_coll\_type\_t. It supports three types of collective operations: (a) UCC\_{ALLTOALL,ALLTOALLV, ALLGATHER, ALLGATHERV, ALLREDUCE, REDUCE_SCATTER, REDUCE_SCATTERV, BARRIER} operations where all participants contribute to the results and receive the results (b) UCC\_{REDUCE, GATHER, GATHERV, FANIN} where all participants contribute to the result and one participant receives the result. The participant receiving the result is designated as root. (c) UCC\_{BROADCAST, SCATTER, SCATTERV, FANOUT} where one participant contributes to the result, and all participants receive the result. The participant contributing to the result is designated as root.
+**Collective Type**: The collective operation supported by UCC is defined by the
+enumeration ucc\_coll\_type\_t. The semantics are briefly described here,
+however in most cases it agrees with the semantics of collective operations in
+the popular programming models such as MPI and OpenSHMEM. When they differ, the
+semantics changes are documented. All collective operations operate on the team.
+For the collective operations defined in the enumeration, all participants of
+the team participate in the collective operations. Further the team should be
+created with endpoints, where the “eps” should be ordered and contiguous.
 
-\b Reduction types: The reduction operation supported by UCC\_{ALLREDUCE, REDUCE} operation is defined by the enumeration ucc\_op\_t. The valid datatypes for the reduction is defined by the enumeration ucc\_datatype\_t.
+UCC supports three types of collective operations: (a) UCC\_{ALLTOALL,ALLTOALLV,
+ALLGATHER, ALLGATHERV, ALLREDUCE, REDUCE_SCATTER,REDUCE_SCATTERV, BARRIER}
+operations where all participants contribute to the results and receive the
+results (b) UCC\_{REDUCE, GATHER, GATHERV, FANIN} where all participants
+contribute to the result and one participant receives the result. The
+participant receiving the result is designated as root. (c) UCC\_{BROADCAST,
+SCATTER, SCATTERV, FANOUT} where one participant contributes to the result, and
+all participants receive the result. The participant contributing to the result
+is designated as root.
 
-\b Ordering: The team can be configured for ordered collective operations or unordered collective operations. For unordered collectives, the user is required to provide the “tag”, which is an unsigned 64-bit integer. 
++ The UCC\_COLL\_TYPE\_BCAST operation moves the data from the src buffer of
```

Comment:
lets remove "src" and "dest" words here, because it is a bit misleading. In coll_args for bcast we init args.src.info.buffer for ALL ranks, root and non root (args.dst.info is not initialized).